iree/turbine/kernel/_support/indexing.py

```diff
@@ -411,18 +411,23 @@ class IndexSequence:
     stride: Optional[IndexExpr | int] = 1
 
     def _subs(
-        self, value: int | IndexExpr, map: dict[IndexSymbol, int]
+        self, value: int | IndexExpr, map: dict[IndexExpr, IndexExpr]
     ) -> int | IndexExpr:
-        new_value = value
-        if isinstance(value, IndexExpr):
-            new_value = value.subs(map)
-        return new_value
+        if isinstance(value, (sympy.Basic, IndexSequence)):
+            return value.subs(map)
+        return value
 
-    def subs(self, map: dict[IndexSymbol, int]):
+    def subs(self, map: dict[IndexExpr, IndexExpr]):
         start = self._subs(self.start, map)
         size = self._subs(self.size, map)
         stride = self._subs(self.stride, map)
         return IndexSequence(start, size, stride)
 
+    def apply_expr(self, symbol: IndexExpr, expr: IndexExpr):
```

Comment:
Should we add some checks here to ensure that the size > 0 and stride is not negative?

Reply:
I wanted to have this method as straightforward as possible. Any pre/post conditions should be enforced on caller side (and `subs` doesn't do any checks either).